2019Labs/CustomSecurityContent/documentation/lab2_openscap.adoc

```diff
@@ -57,103 +76,102 @@ CVRF Version: 1.1
 ----
 
 Notice that this command outputs the OpenSCAP version and versions of supported standards.
-If during your own installation it says `oscap` command not found, you need to install the OpenSCAP tooling since this means that OpenSCAP has not been installed successfully.
 
+== Using ComplianceAsCode content with OpenSCAP command line tool
 
-== Lab 1.1 Security scanning with OpenSCAP
+In this Section we will build the security content for Red Hat Enterprise Linux 8 from ComplianceAsCode source code and then we will use the built content with OpenSCAP command line tool to scan our machine.
 
-. Let's take a look at the compliance content provided by `scap-security-guide`:
+. In ComplianceAsCode source directory, build the content for Red Hat Enterprise Linux 8.
 +
 ----
-rpm -ql scap-security-guide
-...
-/usr/share/xml/scap/ssg/content/ssg-rhel8-cpe-dictionary.xml
-/usr/share/xml/scap/ssg/content/ssg-rhel8-cpe-oval.xml
-/usr/share/xml/scap/ssg/content/ssg-rhel8-ds.xml
-/usr/share/xml/scap/ssg/content/ssg-rhel8-oval.xml
-/usr/share/xml/scap/ssg/content/ssg-rhel8-xccdf.xml
-...
+./build_product rhel8
 ----
-
-. Notice that the provided content in `scap-security-guide` covers a wide range of security baselines such as PCI-DSS, STIG, C2S, Common, CJIS, and others.
-. Notice that content provided in `scap-security-guide` covers a wide range of security baselines.
-For Red Hat Enterprise Linux 8, OSPP (Protection Profile for General Purpose Operating Systems) and PCI-DSS (Payment Card Industry Data Security Standard) profiles are available.
-There are also various formats in which this is provided - human readable HTML guides, SCAP benchmarks, but also Ansible remediation playbooks. 
-
-. Move to the `content` folder so we can avoid typing long paths:
++
+. Take a look at the generated files in the `build` directory.
 +
 ----
-cd /usr/share/xml/scap/ssg/content
+cd build
+ls
 ----
-
-. First, let's check which compliance profiles are available for rhel7.
++
+There are multiple files produced by the build. The file that is going to be used with the OpenSCAP scanner is `ssg-rhel8-ds.xml`. This file is called a SCAP Datastream.
++
+. Check which compliance profiles are available for RHEL8.
 +
 ----
 oscap info ssg-rhel8-ds.xml
 ----
 +
-You will get print out of contents - aforementioned two compliance profiles. OSPP - Protection Profile for General Purpose Operating Systems, and PCI-DSS.
-
-. Next, let's perform our first baseline testing scan with the vanilla OSPP profile.
+You will get a list of profiles contained in the datastream.
+Notice that the provided content in `ssg-rhel8-ds.xml` covers a wide range of security baselines such as PCI-DSS, STIG, C2S, Common, CJIS, and others.
```

Comment:
@matejak Thank you for pointing this out.

This sentence has been moved from the section about listing files contained in scap-security-guide RPM package. This section has been removed because in this lab we work with the upstream ComplianceAsCode project, not with the downstream package.

However, I kept the section about using `oscap info` because I think `oscap info` is a very useful command.
The output of `oscap info ssg-rhel8-ds.xml` will list 7 profiles: cjis, standard, hipaa, ospp, pci-dss, rht-ccp, cui.

I propose to rephrase it this way to be less specific:  

_In the "Profiles:" section you will get a list of profiles contained in the datastream. The datastream contains multiple profiles that cover different security baselines for different purposes. Each profile is identified by a profile ID._

How do you like this ^ proposal?

> downstream has only OSPP and PCI-DSS

I think it's irrelevant because this lab works with upstream ComplianceAsCode project, not  the downstream package shipped in RHEL.

>  We should add that there are those two that are QA'd and the upstream project contains other community-contributed and not tested profiles

None of the profiles in upstream repository are officially tested or supported by Red Hat. All of them are community-contributed. Users that need to use QA'd content should be advised to use content from the RPM package instead of upstream repository.

Reply:
I like `oscap info`, I like your proposal about profiles, and I would also like to have a clarifying statement there, similar to what you have written, i.e. something like:

The built ComplianceAsCode content is available as a RHEL scap-security-guide package. Unlike the upstream repository that we work with now, the package contains only content that is officially tested and supported by Red Hat. Therefore, the scap-security-guide package in RHEL 8 contains only the OSPP and PCI-DSS profiles.

Reply:
@matejak OK, I'll add it.

Reply:
@matejak done